src/Workspaces/Core/Portable/Workspace/ProjectSystem/IFileChangeWatcher.cs

```diff
@@ -66,15 +66,10 @@ public interface IFileChangeContext : IDisposable
         /// Starts watching a file but doesn't wait for the file watcher to be registered with the operating system. Good if you know
         /// you'll need a file watched (eventually) but it's not worth blocking yet.
         /// </summary>
-        IFileWatchingToken EnqueueWatchingFile(string filePath);
-
-        void StopWatchingFile(IFileWatchingToken token);
+        IWatchedFile EnqueueWatchingFile(string filePath);
```

Comment:
I'm keeping the interface there rather than returning IDisposable directly: the comment on this method reminded me we don't have a good way for a caller to know when we did get the watch in place, and we might have to add members for that. And it just keeps the rest of the system a bit more type safe.